Lib/test/test_unittest/testmock/testpatch.py

```diff
@@ -989,12 +989,18 @@ def test_autospec_staticmethod(self):
             Foo.static_method()
             method.assert_called_once_with()
 
+            # Ensure that the original method's signature is checked correctly
+            self.assertRaises(TypeError, Foo.static_method, 42)
+
 
     def test_autospec_classmethod(self):
         with patch('%s.Foo.class_method' % __name__, autospec=True) as method:
             Foo.class_method()
             method.assert_called_once_with()
 
+            # Ensure that the original method's signature is checked correctly
+            self.assertRaises(TypeError, Foo.class_method, 42)
```

Comment:
I think that this test is not enough. Please, create a new test with class and static methods that have some parameters. And check that these parameters do work :)

Reply:
Thanks for reviewing this PR! I added more tests :)